src/fpu/fpu_instructions.h

```diff
@@ -230,28 +224,19 @@ static void FPU_FLD_I32(PhysPt addr,Bitu store_to) {
 	fpu.regs[store_to].d = static_cast<Real64>(blah);
 }
 
-static void FPU_FLD_I64(PhysPt addr,Bitu store_to) {
-	// The range of integers that can fit in the mantissa of a double.
-	// https://en.wikipedia.org/wiki/Double-precision_floating-point_format
-	constexpr int64_t MantissaMin = -(1LL << 53);
-	constexpr int64_t MantissaMax = (1LL << 53);
-
-	FPU_Reg temp_reg;
-	temp_reg.ll = mem_readq(addr);
+static void FPU_FLD_I64(PhysPt addr, Bitu store_to)
+{
+	const int64_t val = mem_readq(addr);
 
-	fpu.regs[store_to].d = static_cast<Real64>(temp_reg.ll);
 	// This is a fix for vertical bars that appear in some games that
-	// use FILD/FIST as a fast 64-bit integer memcpy, such as Carmageddon,
-	// Motor Mash, and demos like Sunflower and Multikolor.
-
-	// If the value won't fit in the mantissa of a double, save the
-	// value into the regs_memcpy register to indicate that the integer
-	// value should be read out by the FPU_FST_I64 function instead.
-	if (temp_reg.ll > MantissaMax || temp_reg.ll < MantissaMin) {
-		fpu.regs_memcpy[store_to] = temp_reg.ll;
-	} else {
-		fpu.regs_memcpy[store_to].reset();
-	}
+	// use FILD/FIST as a fast 64-bit memcpy, such as Carmageddon,
```

Comment:
We should state the problem here too, e.g.:

"This is a fix for the black vertical bars that appear in some games that use the FILD/FIST 64-bit memcpy trick", then examples.

Reply:
> We should state the problem here too, e.g.:
> 
> "This is a fix for the black vertical bars that appear in some games and demos that use the FILD/FIST 64-bit memcpy trick", then the examples.

That is... what I did?

Reply:
Yeah, I just haven't recovered from my brain surgery yet 🤣  Carry on 😎